ocaml/runtime/gc_ctrl.c

```diff
@@ -417,3 +419,73 @@ CAMLprim value caml_ml_runtime_warnings_enabled(value unit)
   CAMLassert (unit == Val_unit);
   return Val_bool(caml_runtime_warnings);
 }
+
+struct gc_tweak {
+  const char* name;
+  uintnat* ptr;
+  uintnat initial_value;
+};
+static struct gc_tweak gc_tweaks[] = {
+  { "custom_work_max_multiplier", &caml_custom_work_max_multiplier, 0 }
+};
+enum {N_GC_TWEAKS = sizeof(gc_tweaks)/sizeof(gc_tweaks[0])};
+
+void caml_init_gc_tweaks(void)
+{
+  for (int i = 0; i < N_GC_TWEAKS; i++) {
+    gc_tweaks[i].initial_value = *gc_tweaks[i].ptr;
+  }
+}
+
+void caml_print_gc_tweaks(void)
+{
+  for (int i = 0; i < N_GC_TWEAKS; i++) {
+    fprintf(stderr, "%s (initial value %ld)\n",
+	gc_tweaks[i].name,
+	gc_tweaks[i].initial_value);
+  }
+}
+
+uintnat* caml_lookup_gc_tweak(const char* name, uintnat len)
+{
+  for (int i = 0; i < N_GC_TWEAKS; i++) {
+    if (strlen(gc_tweaks[i].name) == len &&
+        memcmp(gc_tweaks[i].name, name, len) == 0) {
+      return gc_tweaks[i].ptr;
+    }
+  }
+  return NULL;
+}
+
+CAMLprim value caml_gc_tweak_get(value name)
```

Comment:
Should we add `CAMLparam` to both `caml_gc_tweak_get` and `caml_gc_tweak_set`? I believe the current impls are correct, but it's nice for their correctness to not be sensitive to whether they allocate.

Reply:
Yeah, we absolutely should.

This is a prime example of "code written by people who know too much about the GC", which is a huge source of bugs (if not now, then after the first reasonable-looking refactoring)